dev_tools/check_shellscript_set_options.py

```diff
@@ -12,22 +12,46 @@
     from pathlib import Path
 
 
-def _sets_options_or_is_nolint(line: str) -> bool:
-    return line.strip() in ["set -euxo pipefail", "# nolint(set_options)"]
+def _sets_options_or_is_nolint(line: str, expected_options: str) -> bool:
+    return line.strip() in [ expected_options, "# nolint(set_options)"]
 
 
-def _is_valid_bash_file(filename: Path) -> bool:
-    return any(_sets_options_or_is_nolint(line) for line in filename.open().readlines())
+def _is_valid_shell_file(filename: Path, expected_options: str) -> bool:
+    return any(_sets_options_or_is_nolint(line, expected_options) for line in filename.open().readlines())
```

Comment:
This is generally bad style as the `fd` returned by `filename.open()` is not closed. It might get closed once the GC collects it. It will be closed when the interpreter shuts down. Though here it does not really matter as it is a short-lived python script.

```suggestion
        with filename.open() as fd:
            return any(_sets_options_or_is_nolint(line, expected_options) for line in fd.readlines())
```

Reply:
Not my change, but good point!